### PR TITLE
Handle OS pretty name on old OS without OS release

### LINKED ISSUE #35440: [CI] Broken packaging tests on Centos6
https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+packaging-tests/77/console

Reproduces reliably
```
./gradlew :qa:vagrant:vagrantCentos6#batsPackagingTest
```

```
04:25:49 # Caused by: java.nio.file.NoSuchFileException: /usr/lib/os-release
04:25:49 # 	at sun.nio.fs.UnixException.translateToIOException(UnixException.java:86) ~[?:?]
04:25:49 # 	at sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:102) ~[?:?]
04:25:49 # 	at sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:107) ~[?:?]
04:25:49 # 	at sun.nio.fs.UnixFileSystemProvider.newByteChannel(UnixFileSystemProvider.java:214) ~[?:?]
```

Seems to be triggered by #35352 (730ec1d)

### PR DESCRIPTION
**Pull request by jasontedor**
Some very old ancient versions of Linux do not have /etc/os-release. For example, old Red Hat-like OS. This commit adds a fallback for handling pretty name for these OS.

Closes #35440

### CODE REVIEW

**Review by romseygeek — COMMENTED** [review]
One question about assertions vs exceptions, otherwise LGTM

File: server/src/main/java/org/elasticsearch/monitor/os/OsProbe.java [hunk 1]
```diff
@@ -567,22 +569,33 @@ private String getPrettyName() throws IOException {
...
+     *                     {@code /etc/system-release}
      */
-    @SuppressForbidden(reason = "access /etc/os-release or /usr/lib/os-release")
+    @SuppressForbidden(reason = "access /etc/os-release or /usr/lib/os-release or /etc/system-release")
     List<String> readOsRelease() throws IOException {
         final List<String> lines;
         if (Files.exists(PathUtils.get("/etc/os-release"))) {
             lines = Files.readAllLines(PathUtils.get("/etc/os-release"));
-        } else {
+            assert lines != null && lines.isEmpty() == false;
```
[the remark this answers is not in the document]
Comment (jasontedor) [reply server/src/main/java/org/elasticsearch/monitor/os/OsProbe.java]: Exactly, we code for the platforms that we support and use these assertions to ensure that everything is as expected on those platforms.

File: server/src/main/java/org/elasticsearch/monitor/os/OsProbe.java [hunk 2]
```diff
@@ -567,22 +569,33 @@ private String getPrettyName() throws IOException {
     }
 
     /**
-     * The lines from {@code /etc/os-release} or {@code /usr/lib/os-release} as a fallback. These file represents identification of the
-     * underlying operating system. The structure of the file is newlines of key-value pairs of shell-compatible variable assignments.
+     * The lines from {@code /etc/os-release} or {@code /usr/lib/os-release} as a fallback, with an additional fallback to
+     * {@code /etc/system-release}. These file represents identification of the underlying operating system. The structure of the file is
```
[the remark this answers is not in the document]
Comment (jasontedor) [reply server/src/main/java/org/elasticsearch/monitor/os/OsProbe.java]: I pushed cac0f9f50cb885c43f675a76ee56f7a5a4a2ee57.

File: server/src/main/java/org/elasticsearch/monitor/os/OsProbe.java [hunk 3]
```diff
@@ -567,22 +569,33 @@ private String getPrettyName() throws IOException {
     }
 
     /**
-     * The lines from {@code /etc/os-release} or {@code /usr/lib/os-release} as a fallback. These file represents identification of the
-     * underlying operating system. The structure of the file is newlines of key-value pairs of shell-compatible variable assignments.
+     * The lines from {@code /etc/os-release} or {@code /usr/lib/os-release} as a fallback, with an additional fallback to
+     * {@code /etc/system-release}. These files represents identification of the underlying operating system. The structure of the file is
```
Comment (romseygeek) [inline server/src/main/java/org/elasticsearch/monitor/os/OsProbe.java]: nanonit: s/represents/represent/ 
:)

File: server/src/main/java/org/elasticsearch/monitor/os/OsProbe.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (romseygeek) [reply server/src/main/java/org/elasticsearch/monitor/os/OsProbe.java]: ++

**Review by romseygeek — APPROVED** [review]
LGTM

File: server/src/main/java/org/elasticsearch/monitor/os/OsProbe.java (hunk 3 above)
Comment (jasontedor) [reply server/src/main/java/org/elasticsearch/monitor/os/OsProbe.java]: I pushed c09423d9013165cadf5a4d2f4223d86e10c2f6c3.

Comment (jasontedor) [reply server/src/main/java/org/elasticsearch/monitor/os/OsProbe.java]: Thanks for double-checking. 😄

**Comment by jasontedor** [comment]
@elasticmachine run gradle build tests

**Comment by jasontedor** [comment]
@elasticmachine run gradle builds test

**Comment by jasontedor** [comment]
@elasticmachine run gradle builds tests

**Pull request merged**